aptos-move/framework/src/release_bundle.rs

```diff
@@ -0,0 +1,218 @@
+        std::fs::write(path, bcs::to_bytes(self)?)?;
+        Ok(())
+    }
+
+    /// Constructs a unified error map for all packages in this bundle.
+    pub fn error_mapping(&self) -> ErrorMapping {
+        let mut map = ErrorMapping::default();
+        for pack in &self.packages {
+            let ErrorMapping {
+                mut error_categories,
+                mut module_error_maps,
+            } = bcs::from_bytes::<ErrorMapping>(&pack.package_metadata().error_map)
+                .expect("bcs of error map");
+            map.error_categories.append(&mut error_categories);
+            map.module_error_maps.append(&mut module_error_maps);
+        }
+        map
+    }
+
+    /// Returns a list of all ScriptABIs in this bundle.
+    pub fn abis(&self) -> Vec<ScriptABI> {
+        let mut result = vec![];
+        for pack in &self.packages {
+            let mut abis = pack.abis();
+            result.append(&mut abis);
+        }
+        result
+    }
+
+    /// Returns a list of all module bytecodes in this bundle.
+    pub fn code(&self) -> Vec<&[u8]> {
+        let mut result = vec![];
+        for pack in &self.packages {
+            let mut code = pack.code();
+            result.append(&mut code);
+        }
+        result
+    }
+
+    /// Return a list of CompiledModules in this bundle.
+    pub fn compiled_modules(&self) -> Vec<CompiledModule> {
+        self.code_and_compiled_modules()
+            .into_iter()
+            .map(|(_, c)| c)
+            .collect()
+    }
+
+    /// Return a list of bytecode and CompiledModules in this bundle.
+    pub fn code_and_compiled_modules(&self) -> Vec<(&[u8], CompiledModule)> {
+        self.code()
+            .into_iter()
+            .map(|bc| (bc, CompiledModule::deserialize(bc).unwrap()))
+            .collect()
+    }
+
+    /// Some legacy usages of code require a full copy. This is a helper for those cases.
+    /// TODO: remove unnecessary use of this function
+    pub fn legacy_copy_code(&self) -> Vec<Vec<u8>> {
+        self.code().into_iter().map(|v| v.to_vec()).collect()
+    }
+
+    /// Returns the Move source file names which are involved in this bundle.
+    pub fn files(&self) -> anyhow::Result<Vec<String>> {
+        assert!(
+            !self.source_dirs.is_empty(),
+            "release bundle has no source path information"
+        );
+        let mut result = vec![];
+        for path in &self.source_dirs {
+            let mut files = find_filenames(&[path], |p| extension_equals(p, MOVE_EXTENSION))?;
+            result.append(&mut files);
+        }
+        Ok(result)
+    }
+
+    pub fn named_addresses(&self) -> BTreeMap<String, NumericalAddress> {
+        BTreeMap::new()
+    }
+}
+
+impl ReleasePackage {
+    /// Creates a new released package.
+    pub fn new(package: BuiltPackage) -> anyhow::Result<Self> {
+        Ok(ReleasePackage {
+            metadata: package.extract_metadata(/* TODO: remove */ UpgradePolicy::compat())?,
```

Comment:
Should we remove?  What's the context?  It would be great to put in the comment.

Reply:
This need to be done in another PR. Basically this should become part of `Move.toml`.